### ISSUE TITLE
HMR bug with component registration

### ISSUE THREAD

**Issue by allochi** [fuse-box/fuse-box-vue-seed#6]
Hi,

First, thanks for the good work!

I have been trying to track this bug for couple of days now.

It seems when I use a `router` and add a `vue` file that uses another `vue` as a `component` `hmr` stop working, it update the module, and state that in the console, but nothing happens, if I comment the line where I register the component it start working again.

I event tried global component registration and it's the same thing.

So basically I have

```
 layout.vue
 +- router-view
     +- index.vue
          +- table.vue (component)
```
if I register the component `table.vue` inside `index.vue`, `hmr` stop working, once I comment the line to unregister the component `hmr` star working again. I even tried adding empty `vue` component with no script or anything for debugging and still same result.
